docs/deploy_docs.md

```diff
@@ -0,0 +1,28 @@
+# Contract Deploy for Bridge
+
+Each ERC-20 token to be bridged needs to have a corresponding fungible token contract defined on the Fuel chain. These contracts handle messaging with the base layer gateway contract and are responsible for minting and burning the Fuel side token equivalents.
+
+## Dependencies
+
+| dep         | version                                                  |
+| ----------- | -------------------------------------------------------- |
+| Forc        | [v0.31.1](https://fuellabs.github.io/sway/v0.31.1/introduction/installation.html) |
+| Forc-Client | [v0.31.1](https://fuellabs.github.io/sway/v0.31.1/forc/plugins/forc_client/index.html) |
+
+## Configure
+
+Configure the target ERC-20 token by editing the `bridge-fungible-token/Forc.toml` file:
+
+- **LAYER_1_ERC20_GATEWAY**: The address of the gateway contract on the base layer
+- **LAYER_1_TOKEN**: The address of the ERC-20 token contract on the base layer
+- **LAYER_1_DECIMALS**: The decimal value of the base layer ERC-20 token contract
```

Comment:
We might add a comment here about decimal configuration.
If LAYER_1_DECIMALS > DECIMAL, && (LAYER_1_DECIMALS - DECIMALS) > 19, there will be an overflow in the FUel side token contract.
I've just added a check for this condition in the contract and now register a refund if this config condition occurs, but better to pre-empt that particular condition I think.

Reply:
We should be able to handle both of these cases.

`(LAYER_1_DECIMALS - DECIMALS) > 19` can be addressed with the code changes I suggested on [your review](https://github.com/FuelLabs/bridge-fungible-token/pull/35/files#r1022260197). 

`LAYER_1_DECIMALS < DECIMALS` should be easy to handle I just thought we hadn't done it yet. You would just do a "shift_decimals_right" (aka divide) instead of multiply. Again, it should boil down to either calling a "shift_decimals_left" or "shift_decimals_right" function, check if they return an error (overflow/underflow occurred) and then make sure we can cast the result into a u64.

Note: The "negative decimals" issue is how to address when a developer wants to set the L2 decimals to something like -2 meaning 1 token unit on Fuel equals 100 token units on Ethereum (aka you can never hold less than 100 tokens).